hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlCreateIndexTest.java

```diff
@@ -0,0 +1,217 @@
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlCreateIndexTest extends OptimizerTestSupport {
+
+    private static final String MAP_NAME = "map";
+    private IMap<Integer, Integer> map;
+
+    @BeforeClass
+    public static void beforeClass() {
+        initialize(1, null);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        map = instance().getMap(MAP_NAME);
+        for (int i = 0; i < 200; ++i) {
+            map.put(i, i);
+        }
+    }
+
+    @Test
+    public void basicSqlCreateIndexTest_hash() {
+        String indexName = SqlTestSupport.randomName();
+        String sql = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + MAP_NAME + " (this) TYPE HASH";
+        String selectSql = "SELECT * FROM " + MAP_NAME + " WHERE this = 100";
+
+        checkPlan(false, false, selectSql);
+
+        instance().getSql().execute(sql);
+
+        checkPlan(true, false, selectSql);
+    }
+
+    @Test
+    public void basicSqlCreateIndexTest_sorted() {
+        String indexName = SqlTestSupport.randomName();
+        String sql = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + MAP_NAME + " (this) TYPE SORTED";
+        String selectSql = "SELECT * FROM " + MAP_NAME + " ORDER BY this DESC";
+
+        checkPlan(false, true, selectSql);
+
+        instance().getSql().execute(sql);
+
+        checkPlan(true, true, selectSql);
+    }
+
+    @Test
+    public void basicSqlCreateIndexTest_bitmap() {
+        assertThat(mapContainer(map).getIndexes().getIndex(MAP_NAME)).isNull();
+
+        String indexName = SqlTestSupport.randomName();
+        String sql = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + MAP_NAME + " (__key) TYPE BITMAP " +
+                "OPTIONS ('unique_key' = '__key' , 'unique_key_transformation' = 'OBJECT'); ";
+
+        instance().getSql().execute(sql);
+
+        assertThat(mapContainer(map).getIndexes().getIndex(indexName)).isNotNull();
+    }
+
+    @Test
+    public void sqlCreateIndexWithDefaultTypeTest() {
+        String indexName = SqlTestSupport.randomName();
+        String sql = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + MAP_NAME + " (this)";
+        String selectSql = "SELECT * FROM " + MAP_NAME + " ORDER BY this DESC";
+
+        checkPlan(false, true, selectSql);
+
+        instance().getSql().execute(sql);
+
+        checkPlan(true, true, selectSql);
+    }
+
+    @Test
+    public void indexArgsCannotBeDuplicatedTest() {
+        String sql = "CREATE INDEX IF NOT EXISTS idx ON " + MAP_NAME + " (this, this) TYPE BITMAP";
+        assertThatThrownBy(() -> instance().getSql().execute(sql))
+                .hasMessageContaining("specified more than once");
+    }
+
+    @Test
+    public void indexExistsTest() {
```

Comment:
There should also be inverse test: that the IF NOT EXISTS does nothing, if the index already exists.